pkg/syncer/volume_health_reconciler.go

```diff
@@ -209,55 +308,55 @@ func (rc *volumeHealthReconciler) syncPVC(key string) error {
 	log.Infof("syncPVC: Found Supervisor Cluster PVC: %s/%s", svcPVC.Namespace, svcPVC.Name)
 
 	// The input PVC is a Supervisor Cluster PVC
-	// Find Tanzu Kubernetes Grid PV
-	// Find Tanzu Kubernetes Grid PVC accordingly and update it
-	tkgPV, err := rc.findTKGPVforSupervisorPVC(ctx, svcPVC)
+	// Find list of Tanzu Kubernetes Grid PVs referencing this PVC
+	// Find corresponding Tanzu Kubernetes Grid PVCs and update them
+	tkgPVList, err := rc.findTKGPVforSupervisorPVC(ctx, svcPVC)
 	if err != nil {
 		return err
 	}
-	if tkgPV == nil {
-		log.Debugf("Tanzu Kubernetes Grid PV not found for Supervisor PVC %s/%s", svcPVC.Namespace, svcPVC.Name)
-		return fmt.Errorf("tanzu Kubernetes Grid PV not found for Supervisor PVC %s/%s", svcPVC.Namespace, svcPVC.Name)
+	if tkgPVList == nil {
+		// If no PV is found, the SV PVC may not be referenced within this TKG, so we do not requeue this request.
+		log.Debugf("Tanzu Kubernetes Grid PV not found for Supervisor PVC %s/%s. Igonoring ...", svcPVC.Namespace, svcPVC.Name)
+		return nil
```

Comment:
Line 220, which returns error, is there to fix a health annotation bug with static provisioning.  Have you tested static provisioning with this fix?

Reply:
+1

Reply:
@xing-yang thanks for pointing this out! Spoke to Xing offline and looks like the static provisioning use case will need a design change if we dont want to queue _every_ SV PVC in a TKG. Reverting this to the previous logic and keeping this PR's change to handle the performance issue.